ext/session/session.c

```diff
@@ -2344,15 +2778,61 @@ static PHP_FUNCTION(session_start)
 }
 /* }}} */
 
-/* {{{ proto bool session_destroy(void)
+
+/* {{{ proto array session_info(void)
+   Return session internal data information */
+static PHP_FUNCTION(session_info)
+{
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+
+	if (PS(session_status) != php_session_active) {
+		php_error_docref(NULL, E_WARNING, "Session is not active");
+		RETURN_FALSE;
```

Comment:
I understand what you mean, but session module returns FALSE for invalid session status. I don't mind much change them to NULL including other places, but some users may use "===" for return values. So this would be not compatible change in 7.1.